addons/mail/static/src/js/rotting_mixin/rotting_progress_bar_hook.js

```diff
@@ -0,0 +1,46 @@
+export const rottingProgressBarPatch = {
+    rotIsFiltered: {},
+    async toggleFilterRotten(group) {
+        if (!this.rotIsFiltered[group.id]) {
+            await this.setFilterRotten(group);
+        } else {
+            await this.unsetFilterRotten(group);
+        }
+        group.model.notify();
+    },
+    async setFilterRotten(group) {
+        await group.applyFilter([["is_rotting", "=", true]]);
+        this.rotIsFiltered[group.id] = group;
+        if (this.activeBars[group.serverValue]) {
+            delete this.activeBars[group.serverValue];
+        }
+    },
+    async unsetFilterRotten(group) {
+        await group.applyFilter(undefined);
+        delete this.rotIsFiltered[group.id];
+    },
+    /**
+     * @override
+     */
+    async selectBar(groupId, bar) {
+        if (this.rotIsFiltered[groupId]) {
+            delete this.rotIsFiltered[groupId];
+        }
+        return super.selectBar(groupId, bar);
+    },
+    /**
+     * @override
+     */
+    getGroupCount(group) {
+        if (this.rotIsFiltered[group.id]) {
+            let count = 0;
+            for (const record of group.list.records) {
+                if (record.data.is_rotting) {
+                    count++;
+                }
+            }
+            return count;
```

Comment:
this looks like `.filter(...).length` (or `.reduce()` but that one is less clear for many people)